dask/array/tests/test_reductions.py

```diff
@@ -45,30 +45,35 @@ def reduction_1d_test(da_func, darr, np_func, narr, use_dtype=True):
         assert eq(da_func(darr, dtype='f8'), np_func(narr, dtype='f8'))
         assert eq(da_func(darr, dtype='i8'), np_func(narr, dtype='i8'))
         assert same_keys(da_func(darr, dtype='i8'), da_func(darr, dtype='i8'))
+    if max_leaves:
+        assert eq(da_func(darr, max_leaves=max_leaves), np_func(narr))
+        assert eq(da_func(darr, keepdims=True, max_leaves=max_leaves),
+                  np_func(narr, keepdims=True))
 
 
 def test_reductions_1D_float():
     x = np.arange(5).astype('f4')
     a = da.from_array(x, chunks=(2,))
+    max_leaves = {0: 2}
```

Comment:
All reductions are checked both with and without `max_leaves (although not for *all* cases), as seen in `reduction_2d_test`. A `pytest.mark.parametrize` would probably be cleaner though, rather than manually doing it.